File: 3415-check-if-grid-satisfies-conditions/check-if-grid-satisfies-conditions.java
class Solution {
    public boolean satisfiesConditions(int[][] grid) {
        int m = grid.length;
        int n = grid[0].length;
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                // Check condition with the cell below
                if (i < m - 1 && grid[i][j] != grid[i + 1][j]) {
                    return false;
                }
                // Check condition with the cell to the right
                if (j < n - 1 && grid[i][j] == grid[i][j + 1]) {
                    return false;
                }
            }
        }
        
        return true;
    }
}

File: 3415-check-if-grid-satisfies-conditions/check-if-grid-satisfies-conditions.cpp
class Solution {
public:
    bool satisfiesConditions(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        
        for(int i = 0; i < n; ++i){

            if(i > 0 && grid[0][i] == grid[0][i - 1]) return false;

            for(int j = 1; j < m; ++j){
                if(grid[j][i] != grid[j - 1][i]) return false;
            }
        }

        return true;
    }
};